### ISSUE TITLE
no-extra-parens doesn't work as expected on JSX

### ISSUE THREAD

**Issue by quinn** [eslint/eslint#10568]
I have a repo that demonstrates this error:
https://github.com/quinn/test-eslint-no-parens

**Tell us about your environment**

* **ESLint Version:** v5.0.1
* **Node Version:** v10.2.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": [
        "react"
    ],
    "rules": {
        "no-extra-parens": ["error", "all", { ignoreJSX: "none" }],
    }
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class MyComponent extends React.Component {
	render () {
		return (
			<div>Extra Parens</div>
		)
	}
}

```

```bash
./node_modules/.bin/eslint ./component.js
```

**What did you expect to happen?**
An error about the extra parens.

**What actually happened? Please include the actual, raw output from ESLint.**
(empty output, exit code zero)

**Comment by platinumazure** [comment]
Hi @quinn, thanks for the issue!

I'm not sure those are extra parentheses. With how [ASI](https://stackoverflow.com/questions/2846283/what-are-the-rules-for-javascripts-automatic-semicolon-insertion-asi) works in JavaScript, removing the parentheses leads to the equivalent of this code:

```jsx
class MyComponent extends React.Component {
	render () {
		return;
			<div>Extra Parens</div>;
	}
}
```

So I would hope the rule would not report in this case.

On the other hand, something like this should be reported:

```jsx
class MyComponent extends React.Component {
	render () {
		return (<div>Extra Parens</div>)   // Unnecessary parens for sure
	}
}
```

And in [this demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogW2Vycm9yLCBhbGwsIHsgaWdub3JlSlNYOiBub25lIH1dICovXHJcbmNsYXNzIE15Q29tcG9uZW50IGV4dGVuZHMgUmVhY3QuQ29tcG9uZW50IHtcclxuXHRyZW5kZXIgKCkge1xyXG5cdFx0cmV0dXJuICg8ZGl2PkV4dHJhIFBhcmVuczwvZGl2PikgICAvLyBVbm5lY2Vzc2FyeSBwYXJlbnMgZm9yIHN1cmVcclxuXHR9XHJcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7InByZWZlci1jb25zdCI6Mn0sImVudiI6e319fQ==), it is indeed reported.

Hope this helps!

**Comment by quinn** [comment]
@platinumazure thanks, this is definitely a style / formatting thing rather than needless parens.. I'll see if I can come up with a rule for enforcing the formatting of return statements. thanks for your advice!
